### PR TITLE
[BugFix] iceberg filestatistic Deduplication error

### PR DESCRIPTION
**Pull request by zombee0**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/cost/IcebergStatisticProvider.java [hunk 1]
```diff
@@ -96,13 +96,13 @@ private IcebergFileStats generateIcebergFileStats(IcebergTable icebergTable, Sca
                 if (dataFile.recordCount() == 0) {
                     continue;
                 }
+                if (files.contains(dataFile.path().toString())) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/cost/IcebergStatisticProvider.java:99]: IcebergFileStats icebergFileStats = new IcebergFileStats(0) at line 77

**Review by stephen-shelby — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/cost/IcebergStatisticProvider.java [hunk 2]
```diff
@@ -96,13 +96,13 @@ private IcebergFileStats generateIcebergFileStats(IcebergTable icebergTable, Sca
...
+                    continue;
+                }
+                files.add(dataFile.path().toString());
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/cost/IcebergStatisticProvider.java:102]: If recordCount() is always 0, IcebergFileStats may be null

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/cost/IcebergStatisticProvider.java:102]: done

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**